src/library/Text/Text.js

```diff
@@ -0,0 +1,165 @@
+/* @flow */
+import React, { Component } from 'react';
+import { createStyledComponent } from '../styles';
+
+type Props = {
+  /** Available horizontal alignments */
+  align?: 'start' | 'end' | 'center' | 'justify',
+  /** Available styles */
+  appearance?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'mouse' | 'prose',
+  /** Rendered content */
+  children: React$Node,
+  /** Color of text */
+  color?: string,
+  /** The rendered HTML element, e.g. `'span'`, `'strong'` */
+  element?: string,
+  /** Available font weights */
+  fontWeight?: 'regular' | 'semiBold' | 'bold' | 'extraBold' | number,
+  /** Remove top & bottom margins */
+  noMargins?: boolean
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  Text_backgroundColor_selected: baseTheme.color_theme_40,
+  Text_color: baseTheme.color_text,
+  Text_color_h1: baseTheme.color_gray_100,
+  Text_color_h2: baseTheme.color_gray_80,
+  Text_color_h3: baseTheme.color_gray_80,
+  Text_color_h4: baseTheme.color_gray_80,
+  Text_color_h5: baseTheme.color_gray_100,
+  Text_color_h6: baseTheme.color_gray_80,
+  Text_fontSize: baseTheme.fontSize_ui,
+  Text_fontSize_h1: baseTheme.fontSize_h1,
+  Text_fontSize_h2: baseTheme.fontSize_h2,
+  Text_fontSize_h3: baseTheme.fontSize_h3,
+  Text_fontSize_h4: baseTheme.fontSize_h4,
+  Text_fontSize_h5: baseTheme.fontSize_h5,
+  Text_fontSize_h6: baseTheme.fontSize_h6,
+  Text_fontSize_mouse: baseTheme.fontSize_mouse,
+  Text_fontSize_prose: baseTheme.fontSize_prose,
+  Text_fontWeight_h1: baseTheme.fontWeight_extraBold,
+  Text_fontWeight_h2: baseTheme.fontWeight_bold,
+  Text_fontWeight_h3: baseTheme.fontWeight_bold,
+  Text_fontWeight_h4: baseTheme.fontWeight_bold,
+  Text_fontWeight_h5: baseTheme.fontWeight_bold,
+  Text_fontWeight_h6: baseTheme.fontWeight_regular,
+  Text_lineHeight: baseTheme.lineHeight_prose,
+  Text_lineHeight_heading: baseTheme.lineHeight,
+  Text_lineHeight_headingSmall: baseTheme.lineHeight_prose,
+  Text_marginBottom: baseTheme.space_stack_md,
+  Text_marginBottom_heading: baseTheme.space_stack_sm,
+
+  ...baseTheme
+});
+
+const styles = {
+  root: ({
+    align,
+    appearance: propAppearance,
+    color,
+    element,
+    fontWeight,
+    noMargins,
+    theme: baseTheme
+  }) => {
+    let theme = componentTheme(baseTheme);
+    const headingValues = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
```

Comment:
`headingValues` could be defined outside of this function so it is only created once.